uaclient/util.py

```diff
@@ -416,16 +418,27 @@ def _subp(
     @raises subprocess.TimeoutError when timeout specified and the command
         exceeds that number of seconds.
     """
+    if data is None:
+        # using devnull assures any reads get null, rather
+        # than possibly waiting on input.
+        stdin = open(os.devnull)
+    else:
+        data = data if isinstance(data, bytes) else data.encode("utf-8")
+        stdin = subprocess.PIPE  # type: ignore
```

Comment:
This may go away/change with `subprocess.DEVNULL`, but could we give the initial definition the right Union type definition instead of ignoring the error?

Reply:
Dropped from this branch. but  next time grow a param that requires stdin for subp we can take that approach